runtime/error.c

```diff
@@ -148,13 +165,66 @@ void _bal_error_backtrace_print(ErrorPtr ep) {
     fflush(stderr);
 }
 
+static void processPCs(GC PC* pcs, uint32_t nPCs, uint32_t start, int64_t lineNumber, BacktraceError *error) {
+    // Handle the starting pc seperately
+    processInitialPC(pcs[start], lineNumber, error);
+    if (error->code == NO_DEBUG_INFO) {
+        for (uint32_t i = start + 1; i < nPCs; i++) {
+            // We do not need to pass error_callback because, already we have an error.
+            // Also we do not need to pass data because we do not use it
+            backtrace_syminfo(state, pcs[i], printMissingSymbols, NULL, NULL);
+        }
+    }
+    else {
+        for (uint32_t i = start + 1; i < nPCs; i++) {
+            backtrace_pcinfo(state, pcs[i], printBacktraceLine, storeBacktraceError, error);
+        }
+    }
+}
+
+static void processInitialPC(PC pc, int64_t lineNumber, BacktraceError *error) {
```

Comment:
Don't have two separate callbacks for first line vs other lines. Instead always pass in a line number, with zero meaning no information is being provided.

Reply:
```
// Implementation of backtrace_full_callback
static int printBacktraceLine(void *data, PC pc, const char *filename, int lineno, const char *function) {
    if (lineno != 0) {
        BacktraceStartLine *backtraceStartLine = data;
        lineno = backtraceStartLine->lineNumber;
    }
    printLine(filename, lineno, function);
    return SUCCESS;
}
```


For the `backtrace_pcinfo` case that is true. We can do something like the above for all lines. 

But `backtrace_syminfo` case is a bit different. For the first line, if we get errnum -1, we can get the function name. Since we have the first lineNumber we can generate the first trace line with the function name and lineNumber.  For rest of the lines, lineNumber should be zero. If we pass lineNumber to the callback of `backtrace_syminfo`, we do not have information in that callback to compare the lineNumber as shown above. Therefore `backtrace_syminfo` case we will need two seperate callbacks for first lines vs other lines.

Reply:
Can you pass in a zero line number after the first line?

Reply:
Please ignore my comment. This can be done.

Reply:
Fixed